### PR TITLE
Add Licenses Implementation 

### PR DESCRIPTION
**Pull request by IamHaddad**
Still in Progress

### CODE REVIEW

File: domain/src/commonMain/kotlin/com/islamversity/domain/repo/LicensesRepo.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    fun getAllLicenses(context: CoroutineContext = Dispatchers.Default): Flow<List<LicensesRepoModel>>
+
+}
+
+class LicensesRepoImpl : LicensesRepo {
+    override fun getAllLicenses(context: CoroutineContext): Flow<List<LicensesRepoModel>> {
+        return flow {
+
+            listOf(LicensesRepoModel("s", "s"))
```
Comment (helloseyedjafari) [inline domain/src/commonMain/kotlin/com/islamversity/domain/repo/LicensesRepo.kt]: it has to be "emitted".

File: daggercore/src/main/java/com/islamversity/daggercore/modules/domain/LicensesRepoModule.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.islamversity.daggercore.modules.domain
+
+import com.islamversity.domain.repo.LicensesRepo
+import com.islamversity.domain.repo.LicensesRepoImpl
+import dagger.Module
+import dagger.Provides
+
+@Module
+object LicensesRepoModule {
```
Comment (helloseyedjafari) [inline daggercore/src/main/java/com/islamversity/daggercore/modules/domain/LicensesRepoModule.kt:9]: this module has to be added to `DomainModule` in order to work

**Review by helloseyedjafari — COMMENTED** [review]
well done almost done

File: navigation/src/commonMain/kotlin/com/islamversity/navigation/Screens.kt [hunk 3]
```diff
@@ -85,4 +85,11 @@ sealed class Screens(
             const val url = "/settings"
         }
     }
+
+    class Licenses : Screens(name,url) {
+        companion object {
+            const val name = "com.islamversity.settings.LicensesView"
+            const val url = "/settings"
```
Comment (helloseyedjafari) [inline navigation/src/commonMain/kotlin/com/islamversity/navigation/Screens.kt]: url has to be unique for every "screen",

you can use something like:
`/licenses`

File: settings/src/androidMain/kotlin/com/islamversity/settings/LicensesView.kt [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import javax.inject.Inject
+import android.content.Intent
+
+
+class LicensesView : CoroutineView<ViewLicensesBinding, LicensesState, LicensesIntent>() {
+    @Inject
+    override lateinit var presenter: MviPresenter<LicensesIntent, LicensesState>
+    @Inject
+    lateinit var licenseMapper : Mapper<LicensesUIModel, LicenseItemModel>
+    private val intentChannel = BroadcastChannel<LicensesIntent>(Channel.BUFFERED)
```
Comment (helloseyedjafari) [inline settings/src/androidMain/kotlin/com/islamversity/settings/LicensesView.kt]: if you need this please consider using `CoroutineView.intents`
it's for the same purpose

File: settings/src/commonMain/kotlin/com/islamversity/settings/licenses/LicensesState.kt [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package com.islamversity.settings.licenses
+
+import com.islamversity.core.mvi.BaseState
+import com.islamversity.core.mvi.BaseViewState
+import com.islamversity.settings.models.LicensesUIModel
+
+data class LicensesState(
+    override val base: BaseState,
+    val licenses: List<LicensesUIModel>
```
Comment (helloseyedjafari) [inline settings/src/commonMain/kotlin/com/islamversity/settings/licenses/LicensesState.kt:9]: preferably `private`

File: settings/src/commonMain/kotlin/com/islamversity/settings/settings/SettingsProcessor.kt [hunk 6]
```diff
@@ -216,4 +223,12 @@ class SettingsProcessor(
                 emit(SettingsResult.TranslateFontSize(it.size))
             }
     }
+
+    private val LicensesClicked: FlowBlock<SettingsIntent, SettingsResult> = {
```
Comment (helloseyedjafari) [inline settings/src/commonMain/kotlin/com/islamversity/settings/settings/SettingsProcessor.kt]: please follow variable naming convention:
`licensesClicked`

**Review by helloseyedjafari — COMMENTED** [review]
Thank you so much

**Review by helloseyedjafari — APPROVED** [review]

File: settings/src/androidMain/kotlin/com/islamversity/settings/LicensesView.kt (hunk 4 above)
Comment (IamHaddad) [reply settings/src/androidMain/kotlin/com/islamversity/settings/LicensesView.kt]: Can you please explain more?
I couldn't find anything related

File: settings/src/commonMain/kotlin/com/islamversity/settings/licenses/LicensesState.kt (hunk 5 above)
Comment (IamHaddad) [reply settings/src/commonMain/kotlin/com/islamversity/settings/licenses/LicensesState.kt:9]: Can you please explain to me more?
I need to access it from LicesnsesView and nowhere in the code a State Class property is set to private

File: settings/src/androidMain/kotlin/com/islamversity/settings/LicensesView.kt (hunk 4 above)
Comment (helloseyedjafari) [reply settings/src/androidMain/kotlin/com/islamversity/settings/LicensesView.kt]: you did not use intentChannel anywhere so I said 
please remove this variable.

plus if you need a channel like this please don't create the variable yourself there is already one in the parent class:
`intents`

File: settings/src/commonMain/kotlin/com/islamversity/settings/licenses/LicensesState.kt (hunk 5 above)
Comment (helloseyedjafari) [reply settings/src/commonMain/kotlin/com/islamversity/settings/licenses/LicensesState.kt:9]: Yeah my bad sorry
you are 100% correct

**Pull request merged**